nomad/structs/structs.go

```diff
@@ -1821,7 +1861,7 @@ type Node struct {
 	// SecretID is an ID that is only known by the Node and the set of Servers.
 	// It is not accessible via the API and is used to authenticate nodes
 	// conducting privileged activities.
-	SecretID string `json:"-"`
+	SecretID string
```

Comment:
added this in https://github.com/hashicorp/nomad/pull/10202/files#diff-297649e31d8e88fd89ff63dac7aff450569cca19009616f08c1d73d1e8239725R1824, but don't need it anymore because we're explicitly sanitizing `structs.Node` objects before serializing.